salt/modules/boto_cloudfront.py

```diff
@@ -0,0 +1,441 @@
+# -*- coding: utf-8 -*-
+'''
+Connection module for Amazon CloudFront
+
+.. versionadded:: develop
+
+:depends: boto3
+
+:configuration: This module accepts explicit AWS credentials but can also
+    utilize IAM roles assigned to the instance through Instance Profiles or
+    it can read them from the ~/.aws/credentials file or from these
+    environment variables: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY.
+    Dynamic credentials are then automatically obtained from AWS API and no
+    further configuration is necessary. More information available at:
+
+    .. code-block:: text
+
+        http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/
+            iam-roles-for-amazon-ec2.html
+
+        http://boto3.readthedocs.io/en/latest/guide/
+            configuration.html#guide-configuration
+
+    If IAM roles are not used you need to specify them either in a pillar or
+    in the minion's config file:
+
+    .. code-block:: yaml
+
+        cloudfront.keyid: GKTADJGHEIQSXMKKRBJ08H
+        cloudfront.key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+
+    A region may also be specified in the configuration:
+
+    .. code-block:: yaml
+
+        cloudfront.region: us-east-1
+
+    If a region is not specified, the default is us-east-1.
+
+    It's also possible to specify key, keyid and region via a profile, either
+    as a passed in dict, or as a string to pull from pillars or minion config:
+
+    .. code-block:: yaml
+
+        myprofile:
+            keyid: GKTADJGHEIQSXMKKRBJ08H
+            key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+            region: us-east-1
+'''
+# keep lint from choking on _get_conn and _cache_id
+# pylint: disable=E0602
+
+# Import Python libs
+from __future__ import absolute_import
+import logging
+
+# Import Salt libs
+import salt.ext.six as six
+from salt.utils.odict import OrderedDict
+
+import yaml
+
+# Import third party libs
+try:
+    # pylint: disable=unused-import
+    import boto3
+    import botocore
+    # pylint: enable=unused-import
+    logging.getLogger('boto3').setLevel(logging.CRITICAL)
+    HAS_BOTO = True
+except ImportError:
+    HAS_BOTO = False
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load if boto3 libraries exist.
+    '''
+    if not HAS_BOTO:
+        return False
```

Comment:
Please add a reason behind why the module didn't load:
```
return False, 'The boto_cloudfront modules could not be loaded: boto3 libraries not found.'
```
Or something along those lines?